packages/notification-service/src/blockscout/transfers.ts

```diff
@@ -118,26 +119,39 @@ export function convertWeiValue(value: string) {
     .valueOf()
 }
 
+export function updateProcessedBlocks(transfers: Transfer[], lastBlock: number) {
+  transfers.forEach((transfer) => {
+    if (transfer && processedBlocks.indexOf(transfer.blockNumber) < 0) {
+      processedBlocks.push(transfer?.blockNumber)
+    }
+  })
+  processedBlocks = processedBlocks.filter(
+    (blockNumber) => blockNumber >= lastBlock - MAX_BLOCKS_TO_WAIT
+  )
+}
+
 export async function handleTransferNotifications(): Promise<void> {
   const lastBlockNotified = getLastBlockNotified()
   if (lastBlockNotified < 0) {
     // Firebase not yet ready
     return
   }
+  const blockToQuery = lastBlockNotified - MAX_BLOCKS_TO_WAIT
```

Comment:
We should add the comment made on the PR about blockscout being eventually consistent, which explains why we do this here.